uamqp/error.py

```diff
@@ -191,16 +328,21 @@ class AMQPSessionError(AMQPException):
     """
 
 
-class AMQPLinkError(AMQPException):
+class LinkDetach(AMQPConnectionError):
```

Comment:
So we don't need a generic "LinkError" - we only raise errors on Detach?

Reply:
in general there're two kinds of link level errors:

- 1. errors coming from service which must be via a link detach frame
- 2. errors coming from the client, might be wrong data input, link already closed

to handle errors coming from service, I think LinkDetach is enough (same to Session/Connection as END/CLOSE frame carries the error info).

to handle client side errors, we probably could have this generic "LinkError" as our fallback exception for non-identifies issues and make our error hierarchy extensible.

Reply:
I'd rather not introduce new error types for the sake of it.
We could be consistent with Connection and Session and just name this LinkError rather than LinkDetach. Or we could go the other direction and name the other two ConnectionEnd and SessionClose. In fact, that may not be so bad....

Reply:
I double checked the AMQP spec, I'm on your side now -- I don't want to have a single exception class for each of the condition which seems overkilling.

hierarchy changed back